css-shared-element-transitions/index.bs

```diff
@@ -978,3 +996,20 @@ The {{SameDocumentTransition/finished}} [=getter steps=] are to return [=this's=
                 set |incoming|'s 'object-view-box' property to a value that when applied to |incoming|,
                 will cause the view box to coincide with [=incoming element=]'s [=border box=] in the image.
 </div>
+
+<div algorithm>
+    To <dfn>call the DOM change callback</dfn> of a {{DOMTransition}} |transition|:
+
+    1. [=Assert=]: |transition|'s [=DOMTransition/DOM change callback called=] is false.
+
+    1. Let |callbackPromise| be the result of [=/invoking=] |transition|'s [=DOMTransition/DOM change callback=].
```

Comment:
Why not set DOMTransition/DOM updated promise to |callbackPromise|?

Reply:
That would break this case:

```js
const transition = document.createTransition({
  updateDOM() {
    // …
  }
});

await transition.domUpdated;
console.log('updated!');
```

`'updated!'` would never log, because `transition.domUpdated` never settles. Some other promise that `transition.domUpdated` is later set to is the one that settles.

Reply:
Oh I get it now. Coz we haven't even run the callback yet so we'll end up switching the promise when that happens. SGTM then.